src/native/custom_key_op_handler.c

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include "custom_key_op_handler.h"
+
+static void s_aws_custom_key_op_handler_perform_operation(
+    struct aws_custom_key_op_handler *key_op_handler,
+    struct aws_tls_key_operation *operation) {
+
+    struct aws_jni_custom_key_op_handler *op_handler = (struct aws_jni_custom_key_op_handler *)key_op_handler->impl;
+    AWS_FATAL_ASSERT(op_handler != NULL);
+
+    /* Get the Java ENV */
+    JNIEnv *env = aws_jni_acquire_thread_env(op_handler->jvm);
+    AWS_FATAL_ASSERT(env != NULL);
+
+    jbyteArray jni_input_data = NULL;
+    jobject jni_operation = NULL;
+    bool success = false;
+    bool exception_occured = false;
+
+    if (operation == NULL) {
+        goto clean_up;
```

Comment:
just assert this is true.
it should never happen, let's not add real error handling